src/libraries/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1096,10 +1096,30 @@ private async Task<Stream> InternalGetRequestStream()
             {
                 // We're calling SendRequest with async, because we need to open the connection and send the request
                 // Otherwise, sync path will block the current thread until the request is sent.
+                CancellationTokenSource cts = new();
+                cts.CancelAfter(Timeout);
                 TaskCompletionSource<Stream> getStreamTcs = new();
                 TaskCompletionSource completeTcs = new();
+                cts.Token.Register(() => {
+
+                });
                 _sendRequestTask = SendRequest(async: true, new RequestStreamContent(getStreamTcs, completeTcs));
-                _requestStream = new RequestStream(await getStreamTcs.Task.ConfigureAwait(false), completeTcs);
+                Task<Stream> getStreamTask = getStreamTcs.Task;
+                try
+                {
+                    Task result = await Task.WhenAny((Task)getStreamTask, (Task)_sendRequestTask).ConfigureAwait(false);
+                    if (result == _sendRequestTask)
+                    {
+                        await _sendRequestTask.ConfigureAwait(false); // Propagate the exception
+                        return Stream.Null; // If we successfully completed the request without getting the stream
+                        // We just need to return a null stream, to avoid blocking.
```

Comment:
Nit:
```suggestion
                        // If we successfully completed the request without getting the stream,
                        // return a null stream to avoid blocking.
                        return Stream.Null; 
```